kolibri/plugins/coach/assets/src/views/exams/CreateExamPage/index.vue

```diff
@@ -436,6 +447,27 @@
           );
         }
       },
+      contentIsIndeterminate(content) {
+        if (content.kind === ContentNodeKinds.TOPIC) {
+          const everyExerciseSelected = content.exercises.every(
```

Comment:
Using similar code, you could also filter and count the nodes in `content.exercises` that are in `this.selectedExercises`. Then you can determine the checkbox state on whether the length of that filtered list is 0 (not checked), equal to `content.exercises.length` (checked), or less than `content.exercises.length` (indeterminate)

Reply:
For this kind of scenario, I sometimes like to use lodash/xor to get the difference between sets. e.g. if selected - content.exercises is empty, then checked, etc.